sdk/provisioning/Azure.Provisioning/tests/ProvisioningTests.cs

```diff
@@ -170,6 +171,31 @@ public async Task WebSiteUsingL3SpecificSubscription()
                 }));
         }
 
+        [Test]
+        public async Task WebSiteUsingL3ResourceGroupScope()
+        {
+            var infra = new TestInfrastructure(scope: ConstructScope.ResourceGroup);
```

Comment:
I would have imagined a bool being passed in here `useAnonymouseResourceGroup` or whatever we want to name it.  Basically a flag on the infra instance that determines how it behaves.